subsys/nrf_security/src/drivers/cracen/cracenpsa/src/cracen.c

```diff
@@ -51,7 +51,8 @@ static void cracen_load_microcode(void)
 
 void cracen_acquire(void)
 {
-	nrf_security_mutex_lock(&cracen_mutex);
+	__ASSERT(mbedtls_mutex_lock(&cracen_mutex) == 0,
+		"cracen_mutex not initialized (lock)");
```

Comment:
Separate asserts from the actual action, so that the action is still performed even when asserts are disabled.
e.g.:
```suggestion
	const int lock_ret = mbedtls_mutex_lock(&cracen_mutex);

	__ASSERT(lock_ret == 0, "cracen_mutex not initialized (lock)");
```